buku

```diff
@@ -1456,17 +1464,18 @@ class BukuDb:
             if delay_delete:
                 self.conn.commit()
             else:
-                self.commit_delete()
+                self.commit_delete(retain_order=retain_order)
 
         return True
 
-    def commit_delete(self, apply: bool = True):
+    def commit_delete(self, apply: bool = True, retain_order=False):
         """Commit delayed delete commands."""
         if apply and self._to_delete is not None:
             with self.lock:
                 for id in sorted(set(self._to_delete), reverse=True):
-                    self.delete_rec(id, delay_commit=True, chatty=False)
+                    self.delete_rec(id, delay_commit=True, chatty=False, retain_order=retain_order)
                 self.conn.commit()
+                self.cur.execute('VACUUM')
```

Comment:
`VACUUM` is necessary to shrink the file after deletion (particularly when deleting a large amount of records). Note that it effectively creates (and commits) a transaction of its own, so placing it _before_ `conn.commit()` results in an error.